nd4j/nd4j-serde/nd4j-arrow/src/main/java/org/nd4j/arrow/ByteDecoArrowSerde.java

```diff
@@ -0,0 +1,270 @@
+        else if(dataType.equals(arrow.uint16())) {
+            return org.nd4j.linalg.api.buffer.DataType.UINT16;
+        }
+        else if(dataType.equals(arrow.uint32())) {
+            return org.nd4j.linalg.api.buffer.DataType.UINT32;
+        }
+        else if(dataType.equals(arrow.uint64())) {
+            return org.nd4j.linalg.api.buffer.DataType.UINT64;
+
+        }
+        else if(dataType.equals(arrow.int8())) {
+            return org.nd4j.linalg.api.buffer.DataType.BYTE;
+        }
+        else if(dataType.equals(arrow.int16())) {
+            return org.nd4j.linalg.api.buffer.DataType.SHORT;
+        }
+        else if(dataType.equals(arrow.int32())) {
+            return org.nd4j.linalg.api.buffer.DataType.INT;
+        }
+        else if(dataType.equals(arrow.int64())) {
+            return org.nd4j.linalg.api.buffer.DataType.LONG;
+        }
+        else if(dataType.equals(arrow.float16())) {
+            return org.nd4j.linalg.api.buffer.DataType.HALF;
+        }
+        else if(dataType.equals(arrow.float32())) {
+            return org.nd4j.linalg.api.buffer.DataType.FLOAT;
+        }
+        else if(dataType.equals(arrow.float64())) {
+            return org.nd4j.linalg.api.buffer.DataType.DOUBLE;
+        }
+        else if(dataType.equals(arrow.date32())) {
+            throw new IllegalArgumentException("Unable to convert type " + dataType.name());
+        }
+        else if(dataType.equals(arrow.date64())) {
+            throw new IllegalArgumentException("Unable to convert type " + dataType.name());
+        }
+        else if(dataType.equals(arrow.day_time_interval())) {
+            throw new IllegalArgumentException("Unable to convert type " + dataType.name());
+
+        }
+        else if(dataType.equals(arrow.large_utf8())) {
+            return org.nd4j.linalg.api.buffer.DataType.UTF8;
+        }
+        else if(dataType.equals(arrow.utf8())) {
+            return org.nd4j.linalg.api.buffer.DataType.UTF8;
+        }
+        else if(dataType.equals(arrow.binary())) {
+            return org.nd4j.linalg.api.buffer.DataType.BYTE;
+        }
+        else {
+            throw new IllegalArgumentException("Unable to convert type " + dataType.name());
+        }
+    }
+
+    /**
+     *
+     * @param arrowBuffer
+     * @param dataType
+     * @return
+     */
+    public static DataBuffer fromArrowBuffer(ArrowBuffer arrowBuffer,DataType dataType) {
+        return Nd4j.createBuffer(arrowBuffer,arrowBuffer.capacity(),dataBufferTypeTypeForArrow(dataType));
+    }
+
+    /**
+     * Create a {@link Pair}
+     * of {@link ArrowBuffer} and {@link org.nd4j.linalg.api.buffer.DataType}
+     * based on the input {@link DataBuffer}
+     * @param dataBuffer the input data buffer
+     * @return the pair
+     */
+    public static Pair<ArrowBuffer,DataType> fromNd4jBuffer(DataBuffer dataBuffer) {
+        return Pair.of(new ArrowBuffer(dataBuffer.pointer()),arrowDataTypeForNd4j(dataBuffer.dataType()));
```

Comment:
That does a cast. We need to pass the size as well to call the right constructor:
https://github.com/bytedeco/javacpp-presets/blob/master/arrow/src/gen/java/org/bytedeco/arrow/ArrowBuffer.java

Reply:
@saudet  thanks fixed with the latest commit.